version: '3'

services:

  redis:
    image: redis
    restart: on-failure
    expose:
      - '6379'

  db:
    image: postgres
    restart: on-failure
    env_file:
      - .env
    expose:
      - '5432'
    volumes:
      - pg_data:/var/lib/postgresql/data
    healthcheck:
      test: ['CMD-SHELL', '-c', 'pg_isready -U $POSTGRES_USER']
      interval: 1s
      retries: 2
      timeout: 1s

  app:
    build: .
    tty: true
    ports:
      - '8000:8000'
    command: sh -c 'python manage.py migrate && python manage.py runserver 0.0.0.0:8000'
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - .:/app
    env_file:
      - .env

  celery:
    build: .
    tty: true
    restart: on-failure
    command: celery -A config worker -l INFO -P eventlet
    volumes:
      - .:/app
    depends_on:
      - redis
      - db
      - app
    env_file:
      - .env

  celery-beat:
    build: .
    tty: true
    restart: on-failure
    command: celery -A config beat -l info -S django
    volumes:
      - .:/app
    depends_on:
      - redis
      - db
      - app
    env_file:
      - .env

volumes:
  pg_data: